src/main/java/pl/edu/mimuw/chess/Player.java

```diff
@@ -0,0 +1,161 @@
+package pl.edu.mimuw.chess;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static pl.edu.mimuw.chess.ChessColor.BLACK;
+import static pl.edu.mimuw.chess.ChessColor.WHITE;
+
+public class Player {
+  private List<ChessPiece> pieces;
+  private final ChessColor color;
+
+  public Player(ChessColor color) {
+    this.color = color;
+    this.pieces = initPieces();
+  }
+
+  public ChessColor getColor() {
+    return this.color;
+  }
+
+  public static boolean isMoveCorrect(V2 position1, V2 position2, ChessPiece[][] chessPieces) {
+    if (!Checkboard.isOnBoard(position1) || !Checkboard.isOnBoard(position2))
+      return false;
+    if (chessPieces[position2.x][position2.y] != null) {
+      if (chessPieces[position2.x][position2.y].getColor() == chessPieces[position1.x][position1.y].getColor())
+        return false;
+    }
+    int abs1 = Math.abs(position1.x - position2.x);
+    int abs2 = Math.abs(position1.y - position2.y);
+    //move of knights
+    if (abs1 == 1 && abs2 == 2)
+      return true;
+    if (abs1 == 2 && abs1 == 1)
+      return true;
+    //move horizontal or vertical
+    if (abs1 == 0) {
+      for (int i = Math.min(position1.y, position2.y) + 1; i <= Math.max(position1.y, position2.y) - 1; i++) {
+        if (chessPieces[position1.x][i] != null)
+          return false;
+      }
+      return true;
+    }
+    if (abs2 == 0) {
+      for (int i = Math.min(position1.x, position2.x) + 1; i <= Math.max(position1.x, position2.x) - 1; i++) {
+        if (chessPieces[i][position1.y] != null)
+          return false;
+      }
+      return true;
+    }
+    //incorrect move
+    if (abs1 != abs2)
+      return false;
+    //diagonal
+    int sign1 = 1, sign2 = 1;
+    if (position2.x - position1.x < 0)
+      sign1 = -1;
+    if (position2.y - position1.y < 0)
+      sign2 = -1;
+    var currentPosition = new V2(position1.x + sign1, position1.y + sign2);
+    int counter = 0;
+    //counter in case I did any of the conditions wrong
+    while (currentPosition.x != position2.x && currentPosition.y != position2.y && counter < 10) {
+      counter++;
+      if (chessPieces[currentPosition.x][currentPosition.y] != null)
+        return false;
+      currentPosition = new V2(currentPosition.x + sign1, currentPosition.y + sign2);
+    }
+    return true;
+  }
+
+  public List<ChessPiece> initPieces() {
+    var newPieces = new ArrayList<ChessPiece>();
+
+    if (this.color == WHITE) {
+      newPieces.add(new King(new V2(7, 3), WHITE));
```

Comment:
te dwa kawalki kodu to prawie sparamtryzowane duplikaty, można do metody pomocniczej to wydzielić 😉